### PR TITLE
Check environment at startup

### LINKED ISSUE #62: Add "getting started" tests on first launch
As an enhancement, it might be worthwhile to add a getting started page on the first launch of the package.  I'm thinking of a basic checklist-driven workflow that checks:
- `latexmk` found
- PDF reader found
- basic compilation successful (perhaps include a short tex file in the package?)

### PR DESCRIPTION
**Pull request by yitzchak**
Expands the `checkEnvironment` function to do other checks besides deprecation notices and version migration. Since information checks are done also the messages are placed in the log versus using Atom notifications.

Current checks:
1. `latexmk` version check. Helps warn users about #243 situations. Also verifies that `latexmk` can be run outside of a build. [Error severity]
1. `Rscript` versions check. Mostly just to make sure `Rscript` can be run. [warning severity]
2. `knitr` and `patchSynctex` library check. [warning severity]
3. PDF capable viewer check. [error severity]
4. PS capable viewer check. [warning severity]
5. DVI capable viewer check. [warning severity]

Addresses some of #54 (does not automatically configure TeX path).

Resolves #62.

### CODE REVIEW

File: lib/composer.js [hunk 1]
```diff
@@ -17,7 +17,7 @@ export default class Composer {
     this.configChange = atom.config.onDidChange('latex', () => {
       this.rebuildCompleted = new Set()
     })
-    this.checkEnvironment()
+    if (!atom.inSpecMode()) this.checkEnvironment()
```
Comment (thomasjo) [inline lib/composer.js]: Since this is not an "exit junction" (a `return` statement or similar), let's wrap this one in braces and split it across multiple lines.

File: spec/parsers/log-parser-spec.js [hunk 2]
```diff
@@ -40,18 +40,18 @@ describe('LogParser', () => {
...
+      expect(_.countBy(result.messages, 'type').error).toBe(3)
     })
 
     it('associates an error with a file path, line number, and message', () => {
       const logFile = path.join(fixturesPath, 'errors.log')
       const texFile = path.join(fixturesPath, 'errors.tex')
       const parser = new LogParser(logFile, texFile)
       const result = parser.parse()
-      const error = _.find(result.messages, (message) => { return message.type === 'Error' })
+      const error = _.find(result.messages, (message) => { return message.type === 'error' })
```
Comment (thomasjo) [inline spec/parsers/log-parser-spec.js]: Since we're changing this anyway, maybe we can change this to `Array.find`? But we can also do that in a separate PR at some point (remove as much lodash stuff as possible).

File: lib/builder-registry.js [hunk 3]
```diff
@@ -17,6 +17,14 @@ export default class BuilderRegistry {
     return this.resolveAmbiguousBuilders(candidates, this.getBuilderFromMagic(filePath))
   }
 
+  async checkEnvironment () {
```
Comment (yitzchak) [inline lib/builder-registry.js]: `checkRuntimeDependencies` sounds good.
